apps/framework-cli/src/cli/routines/metrics_console/run_console/app.rs

```diff
@@ -1,26 +1,39 @@
-use super::client::{ParsedMetricsData, PathMetricsData};
-use std::error;
+use prometheus_parse::HistogramCount;
+use ratatui::layout::Rect;
+
+use super::client::{parsing_histogram_data, ParsedMetricsData, PathMetricsData};
+use std::{collections::HashMap, error};
 
 pub type AppResult<T> = std::result::Result<T, Box<dyn error::Error>>;
 
 pub struct App {
+    pub viewport: Rect,
+    pub state: String,
     pub running: bool,
     pub average: f64,
     pub requests_per_sec: f64,
     pub total_requests: f64,
     pub summary: Vec<PathMetricsData>,
     pub starting_row: usize,
+    pub path_detailed_data: Option<Vec<HistogramCount>>,
+    pub path_requests_per_sec: HashMap<String, f64>,
+    pub requests_per_sec_vec: HashMap<String, Vec<u64>>,
 }
 
 impl Default for App {
     fn default() -> Self {
         Self {
+            viewport: Rect::new(0, 0, 0, 0),
+            state: "main".to_string(),
```

Comment:
What's this state? which panel this is on? if that's the case should this be an enum instead?